.github/README.md

```diff
@@ -0,0 +1,392 @@
+  - Modify package.json:
+  ```
+  "name": "@appoptics/apm-binding-dev",
+  "staging_host": "https://apm-appoptics-bindings-node-dev-staging.s3.us-east-1.amazonaws.com",
+  "production_host": "https://apm-appoptics-bindings-node-dev-production.s3.amazonaws.com",
+  ```
+  - Commit updated `package.json` to `master` all branches.
+
+To start fresh on the dev repo run `npm run dev:repo:reset` again.
+
+# Development & Release with GitHub Actions 
+
+> **tl;dr** Push to feature branch. Create Pull Request. Merge Pull Request. Push version tag to release.
+
+## Overview
+
+The package is `node-pre-gyp` enabled and is published in a two step process. First prebuilt add-on tarballs are uploaded to an S3 bucket, and then an NPM package is published to the NPM . Prebuilt tarballs must be versioned with the same version as the NPM package and they must be present in the S3 bucket prior to the NPM package itself being published to the registry.
+
+There are many platforms that can use the prebuilt add-on but will fail to build it, hence the importance of the prebuilts.
+
+## Usage
+
+### Prep - Push Dockerfile
+
+* Push to master is disabled by branch protection.
+* Push to branch which changes any Dockerfile in the `.github/docker-node/` directory will trigger [docker-node.yml](./workflows/docker-node.yml).
+* Workflow will:
+  - Build all Dockerfiles and create a [single package](https://github.com/appoptics/appoptics-bindings-node/pkgs/container/appoptics-bindings-node%2Fnode) named `node` scoped to `appoptics/appoptics-bindings-node` (the repo). Package has multiple tagged images for each of the dockerfiles from which it was built. For example, the image created from a file named `10-centos7-build.Dockerfile` has a `10-centos7-build` tag and can pulled from `ghcr.io/appoptics/appoptics-bindings-node/node:10-centos7-build`. Since this repo is public, the images are also public.
+* Workflow creates (or recreates) images used in other workflows.
+* Manual trigger supported.
+
+```
+push Dockerfile ─► ┌───────────────────┐ ─► ─► ─► ─► ─►
+                   │Build Docker Images│ build & publish
+manual ──────────► └───────────────────┘     
+```
+
+### Develop - Push
+
+* Push to master is disabled by branch protection.
+* Push to branch will trigger [push.yml](./workflows/push.yml). 
+* Workflow will:
+  - Build the code pushed on a default image. (`node` image from docker hub).
+  - Run the tests against the build.
+* Workflow confirms code is not "broken".
+* Manual trigger supported. Enables to select image from docker hub or local files (e.g. `erbium-buster-slim`) to build code on.
+* Naming a branch with `-no-action` ending disables this workflow. Use for documentation branches edited via GitHub UI.
+```
+push to branch ──► ┌───────────────────┐ ─► ─► ─► ─► ─►
+                   │Single Build & Test│ contained build
+manual (image?) ─► └───────────────────┘ ◄── ◄── ◄── ◄──
+```
+
+### Review - Pull Request
+
```

Comment:
A little off topic, but wondering if this workflow should also ignore the `-no-action` branch.  Just noticed that my formatting commit to this PR triggered the `review.yml` workflow: https://github.com/appoptics/appoptics-bindings-node/pull/72/checks

Reply:
tried it.... `on:` `pull_request` can't ignore branch...